packages/engine/src/store/actionCreators/createUpdateInputValues.ts

```diff
@@ -0,0 +1,16 @@
+import { NodeValue, SetterCreator } from '@store/types'
+
+export const createUpdateInputValues: SetterCreator<'updateInputValues'> =
+  (setState) => (inputId: string, value: NodeValue) => {
+    setState((state) => {
+      const input = state.inputs[inputId]
+      if (!input) {
+        //throw new Error(`Input with id ${inputId} not found.`)
+        return
+      }
+
+      input.targetNodeIds.forEach((nodeId) => {
+        state.nodeValues[nodeId] = value
```

Comment:
In the future, we may want some kind of value merging for when multiple inputs target the same value (additive, etc) Would likely want a whole UI for that though, feels pretty far off in terms of needs